File: LES_EXAMENS/EXAM_PYTHON_2025_2026_IA_RATTRAPAGE/exerciceD.c
#include <stdio.h>
int a,n,U0,Ui_1,Ui;
int main(){
    do{
        printf("saisir un entier n>=0: ");
        scanf("%d",&n);
    }while(n<0);
     do{
        printf("saisir un entier a!=0: ");
        scanf("%d",&a);
    }while(n==0);   
    U0=a;
    Ui_1=U0;
    printf("{ %d ",U0);
    for(int j=0;j<n;j++){
        Ui=7*(Ui_1+a);
        printf(" %d ",Ui);
        Ui_1=Ui;
    }
    printf("}\n");
    return 0;
}

File: LES_EXAMENS/EXAM_PYTHON_ET_EXAM_ALGO_C__2025_2026_IA_RATTRAPAGE/exerciceD.py
while True:
    n=int(input("saisir un entier n>=0 "))
    if n>=0:
        break
while True:
    a=int(input("saisir un entier a!=0 "))
    if n!=0:
        break
U0=a
Ui_1=U0
print(f"[ {U0} ",end="")
for i in range(0,n):
    Ui=7*(Ui_1+a)
    print(f" {Ui} ",end="")
    Ui_1=Ui
print("]")